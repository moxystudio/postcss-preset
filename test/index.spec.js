'use strict';

const preset = require('..');

jest.mock('postcss-import', () => (options) => ['postcss-import', options].filter((val) => val));
jest.mock('postcss-url', () => (options) => ['postcss-url', options].filter((val) => val));
jest.mock('postcss-mixins', () => (options) => ['postcss-mixins', options].filter((val) => val));
jest.mock('postcss-css-variables', () => (options) => ['postcss-css-variables', options].filter((val) => val));
jest.mock('postcss-for', () => (options) => ['postcss-for', options].filter((val) => val));
jest.mock('postcss-conditionals', () => (options) => ['postcss-conditionals', options].filter((val) => val));
jest.mock('postcss-cssnext', () => (options) => ['postcss-cssnext', options].filter((val) => val));

afterEach(jest.resetAllMocks);

it('should have a good base config', () => {
    expect(preset()).toMatchSnapshot();
});

it('should pass options.importPath to postcss-import ', () => {
    expect(preset({ importPath: 'foo' })).toMatchSnapshot();
});

it('should pass options.mixinsPath to postcss-mixins ', () => {
    expect(preset({ mixinsPath: 'foo' })).toMatchSnapshot();
});

it('should enable postcss-url if options.url is true', () => {
    expect(preset({ url: true })).toMatchSnapshot();
});

it('should pass options.url to postcss-url', () => {
    expect(preset({ url: { foo: 'bar' } })).toMatchSnapshot();
});

it('should disable postcss-css-variables if options.cssVariables is false', () => {
    expect(preset({ cssVariables: false })).toMatchSnapshot();
});

it('should pass options.cssVariables to postcss-css-variables', () => {
    expect(preset({ cssVariables: { foo: 'bar' } })).toMatchSnapshot();
});

it('should pass options.browsers to postcss-cssnext ', () => {
    expect(preset({ browsers: 'foo' })).toMatchSnapshot();
});
