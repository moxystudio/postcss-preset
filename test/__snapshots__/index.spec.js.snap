// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have a good base config 1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
      Object {
        "path": undefined,
      },
    ],
    Array [
      "postcss-mixins",
      Object {
        "mixinsDir": undefined,
      },
    ],
    "postcss-for",
    Array [
      "postcss-css-variables",
    ],
    "postcss-conditionals",
    Array [
      "postcss-cssnext",
      Object {
        "features": Object {
          "autoprefixer": Object {
            "remove": false,
          },
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "customProperties": false,
        },
      },
    ],
  ],
}
`;

exports[`should pass options.importPath to postcss-import  1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
      Object {
        "path": "foo",
      },
    ],
    Array [
      "postcss-mixins",
      Object {
        "mixinsDir": undefined,
      },
    ],
    "postcss-for",
    Array [
      "postcss-css-variables",
    ],
    "postcss-conditionals",
    Array [
      "postcss-cssnext",
      Object {
        "features": Object {
          "autoprefixer": Object {
            "remove": false,
          },
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "customProperties": false,
        },
      },
    ],
  ],
}
`;

exports[`should pass options.mixinsPath to postcss-mixins  1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
      Object {
        "path": undefined,
      },
    ],
    Array [
      "postcss-mixins",
      Object {
        "mixinsDir": "foo",
      },
    ],
    "postcss-for",
    Array [
      "postcss-css-variables",
    ],
    "postcss-conditionals",
    Array [
      "postcss-cssnext",
      Object {
        "features": Object {
          "autoprefixer": Object {
            "remove": false,
          },
          "browsers": Array [
            "extends browserslist-config-google",
          ],
          "customProperties": false,
        },
      },
    ],
  ],
}
`;
